<launch>
    <arg name="camera_pose_using_camera" default="false"/>
    <arg name="detecting_using_camera" default="false"/>
    <arg name="visualize" default="true"/>
    <arg name="rvizconfig" default="$(find aruco_detection)/rviz/detector.rviz" />
    <param name="camera_pose_ok" value="false"/>

    <node pkg="aruco_detection" type="camera_pose_node.py" name="camera_pose_node" output="log" respawn="true">
        <param name="use_camera" value="$(arg camera_pose_using_camera)"/>
    </node>
    <node pkg="aruco_detection" type="detector_node.py" name="detector_node" output="log" respawn="true">
        <param name="use_camera" value="$(arg detecting_using_camera)"/>
        <param name="visualize" value="$(arg visualize)"/>
    </node>
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />
</launch>
